FRAuth/FRAuth/Callbacks/DeviceBindingCallback.swift

```diff
@@ -0,0 +1,221 @@
+// 
+//  DeviceBindingCallback.swift
+//  FRAuth
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+/**
+ * Callback to collect the device binding information
+ */
+open class DeviceBindingCallback: MultipleValuesCallback {
+    
+    //  MARK: - Properties
+    
+    /// The userId received from server
+    private(set) var userId: String!
+    /// The userName received from server
+    private(set) var userName: String!
+    /// The challenge received from server
+    private(set) var challenge: String!
+    /// The authentication type of the journey
+    private(set) var deviceBindingAuthenticationType: DeviceBindingAuthenticationType!
+    // The title to be displayed in biometric prompt
+    private(set) var title: String!
+    // The subtitle to be displayed in biometric prompt
+    private(set) var subtitle: String!
+    // The description to be displayed in biometric prompt
+    private(set) var promptDescription: String!
+    // The timeout to be to expire the biometric authentication
+    private(set) var timeout: Int?
+    
+    /// Jws input key in callback response
+    private var jwsKey: String!
+    /// Device name input key in callback response
+    private var deviceNameKey: String!
+    /// Device id input key in callback response
+    private var deviceIdKey: String!
+    /// Client Error input key in callback response
+    private var clientErrorKey: String!
+    
+    //  MARK: - Init
+    
+    /// Designated initialization method for DeviceBindingCallback
+    ///
+    /// - Parameter json: JSON object of DeviceBindingCallback
+    /// - Throws: AuthError.invalidCallbackResponse for invalid callback response
+    required public init(json: [String : Any]) throws {
+        
+        guard let callbackType = json[CBConstants.type] as? String else {
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        guard let outputs = json[CBConstants.output] as? [[String: Any]], let inputs = json[CBConstants.input] as? [[String: Any]] else {
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        for output in outputs {
```

Comment:
`can we refactor this code ? we could do some thing like this

   inputs.forEach { input in
            foo(input[CBConstants.name] as? String)
        }
        
        `func` foo(name: String) {
        switch name {
        case Constants.jws: {
            self.jws = name
        }
        }
    }
`